packages/edit-site/src/components/global-styles/ui.js

```diff
@@ -237,6 +237,33 @@ function GlobalStylesBlockLink() {
 	}, [ selectedBlockClientId, selectedBlockName, blockHasGlobalStyles ] );
 }
 
+function GlobalStylesEditorCanvasContainerLink() {
+	const { goTo, location } = useNavigator();
+	const editorCanvasContainerView = useSelect(
+		( select ) =>
+			unlock( select( editSiteStore ) ).getEditorCanvasContainerView(),
+		[]
+	);
+
+	// If the user switches the editor canvas container view, redirect
+	// to the appropriate screen. This effectively allows deep linking to the
+	// desired screens from outside the global styles navigation provider.
+	useEffect( () => {
+		if ( editorCanvasContainerView === 'global-styles-revisions' ) {
+			// Switching to the revisions container view should
+			// redirect to the revisions screen.
+			goTo( '/revisions' );
+		} else if (
+			editorCanvasContainerView === 'style-book' &&
```

Comment:
Thanks for working this one out! I can now switch between views.

Do you think it would be okay to keep it generic, that is, if any container view is open? I'm wondering in case another container view is added down the road.

I tested and it appears to work for this situation.

```suggestion
			!! editorCanvasContainerView &&
```

Reply:
Good call! Yes, keeping it generic will help if and when we add any other container views as this logic should redirect away from the revisions screen if the container view is anything other than revisions and we're on the revisions screen 👍

That appears to work well to me, too, so I've committed in https://github.com/WordPress/gutenberg/pull/51149/commits/e35f61b4e2ca8f3ae0916ce1487a52e0ac5094f7 with a slightly updated comment.